### DISCUSSION TITLE
Why mixing python3 versions?

### DISCUSSION THREAD

**Discussion by alanwilter** [Homebrew/discussions#2565] [Everyday usage]
I use `python3` and the one that `brew` points is `v3.9.9`.

All seemed to be happy until I wanted to use `ipython`, which installs `python 3.10`!!!.

Why that? I found out to my displeasure  surprise that there are other apps using this `python 3.10`: `pipx` and `thefuck`. Those one I don't care but `ipython`???

I'll illustrate with an exemple:

```bash
brew install ipython
brew install open-babel
```

Start `ipython` and try:

``` python
import openbabel
ModuleNotFoundError: No module named 'openbabel'
```
But, with `python3` it will work.

**Bottomline**, `ipython` need to use the same `python3` shipped by default with `brew`. If this is not at least an inconsistency, to not say a bug, then what's the logic behind it?

Is there an alternative to force `ipython` from brew to use same `python 3.9.9`? 

And yes, I know and I did: `brew uninstall ipython` and  `/usr/local/bin/python3 -m pip install ipython`.
However, what's the point then that instead of simply running `bubu` I have to remember to to run `pip install ipython -U` as well?

Or, is it anyhow related to https://github.com/Homebrew/homebrew-core/issues/76621?

**Comment by carlocab** [comment]
We're in the process of migrating all formulae to depend on Python 3.10. There are a lot of those, so it takes some time. You're welcome to help out if you wish.

**Reply by alanwilter** [reply]
Well, I can help, definitely! What do I need to do?

**Reply by suletm** [reply]
It's been a while since the last follow-up on this thread. It's unclear if the help is still needed or not. It would be less-disruptive to finish the migration and only then make the switch of the ipython version to use 3.10, would not it ? The issue is still there with Homebrew 3.5.4.

**Reply by alebcay** [reply]
Hi @suletm, more details are tracked on https://github.com/Homebrew/homebrew-core/pull/87075. You can view all pull requests that have been opened regarding moving to Python 3.10 with https://github.com/Homebrew/homebrew-core/pulls?q=is%3Apr+sort%3Aupdated-desc+label%3Apython-3.10-migration+.

At the time of writing, there have been 481 pull requests with that label, so I would hardly consider such a change "less disruptive". Part of the migration effort has also involved splitting several Python-dependent formulae (each of which has many dependents) into smaller formulae - `glib` (migration effort at https://github.com/Homebrew/homebrew-core/pull/103916) and `python-tabulate` (migration effort at https://github.com/Homebrew/homebrew-core/pull/99064) to make the migration less painful.
